docs/proposals/20201125-jaeger-traces-store.md

```diff
@@ -0,0 +1,124 @@
+
+Follows possible multitenant Jaeger deployment topologies. All deployment topologies assume that Jaeger will be deployed by the [Jaeger Operator](https://github.com/jaegertracing/jaeger-operator) which depends on [OpenShift Elasticserach operator](https://github.com/openshift/elasticsearch-operator) and [Strimzi operator](https://operatorhub.io/operator/strimzi-kafka-operator) if Kafka is used. Jaeger can be as well configured to use externally managed Elasticsearch or Kafka instances.
+
+#### Single Jaeger instance
+
+This deployment topology deploys a single Jaeger instance for all tenants with a single Elasticsearch instance. The data for each tenant would contain a unique label identifying a tenant. The label would be dynamically injected in the Observatorium API service or in the OpenTelemetry collector.
+
+This deployment strategy will not work with the current Jaeger components. Here are few issues. The list might not be complete:
+* Service and operation name API. The service query API nor storage does not expose/store labels.
+* Dependency/service architecture diagram. To support this feature dependency schema, query, collector and Spark aggregation would have to change.
+* Data retention/TTL configurable pre tenant. The data retention in Elasticsearch is configurable per index.
+
+##### OpenTelemetry collector configuration
+
+The OpenTelemetry collector is configured with a single exporter to export data to a single Jaeger collector.
+
+#### Jaeger instance per tenant
+
+This deployment topology deploys a Jaeger instance per tenant and all Jaeger instances use a single Elasticsearch instance. The data for each tenant is stored in a separate Elasticsearch index. Jaeger was designed to separate tenant data per index. Therefore, all Jaeger functionality works well with this deployment strategy.
+
+##### OpenTelemetry collector configuration
+
+The OpenTelemetry collector is configured with an exporter per tenant that exports data to Jaeger collector allocated per a single tenant.
+
+### Jaeger Query
+
+There are two ways users could access data stored in Jaeger:
+1. Expose Jaeger Query API in the Observatorium API.
+2. Expose Jaeger UI outside the cluster.
+
+#### Expose Jaeger Query API
+
+Jaeger query API would be exposed in the Observatorium API. The users would have to deploy Jaeger UI externally and use Observatorium API as a data source.
+
+There are multiple Jaeger query APIs, in this case we propose to expose [api_v3](https://github.com/jaegertracing/jaeger-idl/tree/master/proto/api_v3). This API returns OpenTelemetry compatible payload.
+
+#### Expose Jaeger UI
```

Comment:
Which one is the solution that is being proposed? Expose query API or to expose UI? Are there any additional concerns such as tenant authentication?

Reply:
For this we would like to get feedback from the Observatorium team. 

For us, it would be better to expose just the UI.

Reply:
The Query API **must** be exposed outside the cluster for alternative UIs.

- If the UI is outside the cluster it is somewhat simple to marry it to the token-refresher. 
- If the UI is inside the cluster changes to the UI must be made to include the tenant (a part of the Authorization header the UI will receive) in the query the UI makes to the Query API.  I have concerns that this change would be difficult to audit.  For that reason I believe the Observatorium API should only expose query API, not UI.

Reply:
From my perspective, I think it depends on which use cases are envisioned. If there are alternative UIs, as @esnible mentions, and we want to enable users to use them, then exposing API makes sense. Otherwise, if the Jaeger UI is supposed to be the only way to interact with the storage backend, do we gain anything by having read path in the API?

Just to give full picture, we do both for metrics: Users can interact with the read API, but they can also access the Thanos UI in a browser (provided they can authenticate themselves).

Reply:
I have updated this section. We need to expose query API that the Jaeger UI uses. If we expose the UI directly it depends. I am fine following the current design on exposing only APIs.

Reply:
I think UI would be fine, we do that with Thanos TBH, right @onprem @matej-g @jessicalins ?

Reply:
As I mentioned in my comment above, I believe it depends on the uses we foresee. For Thanos, we expose _both_ UI and query endpoint(s) (but because for metrics it makes sense, depending on use case: whether data is being queried as a data source via the `query` / `query_range` endpoint or a tenant is accessing the Thanos UI in a browser to run queries).